test/language/expressions/assignment/dstr/syntax-error-ident-ref-default-escaped-ext.js

```diff
@@ -0,0 +1,40 @@
+// This file was procedurally generated from the following sources:
+// - src/identifier-names/default-escaped-ext.case
+// - src/identifier-names/default/obj-assignment-identifier.template
+/*---
+description: default is a valid identifier name, using extended escape (IdentiferReference in ObjectAssignmentPattern cannot be a ReservedWord)
+esid: prod-AssignmentPattern
+features: [destructuring-assignment]
+flags: [generated]
+negative:
+  phase: parse
+  type: SyntaxError
+info: |
+    AssignmentPattern:
+      ObjectAssignmentPattern
+
+    ObjectAssignmentPattern:
+      { AssignmentPropertyList }
+
+    AssignmentPropertyList:
+      AssignmentProperty
+      AssignmentPropertyList , AssignmentProperty
+
+    AssignmentProperty:
+      IdentifierReference Initializer_opt
+      PropertyName : AssignmentElement
+
+    IdentifierReference:
+      Identifier
+      [~Yield]yield
+      [~Await]await
+
+    Identifier:
+      IdentifierName but not ReservedWord
+
+---*/
+
+
+$DONOTEVALUATE();
+
+var x = { def\u{61}ult } = { default: 42 };
```

Comment:
and we also capture the error in case it's an identifier reference and not a property name.